x-pack/plugins/fleet/server/services/security/types.ts

```diff
@@ -5,9 +5,16 @@
  * 2.0.
```

Comment:
I redefined all the types to accomodate the new router wrapper. Some of these types are directly lifted from the ones provided by Core for the router. I tried to add comments wherever possible to explain it.